### PR TITLE
V3: remove the api package

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
# Description
Remove the api package and add it directly to request operation.
Refactoring of actual v3 middleware `fn` to request editors.

## Checklist
(For exoscale contributors)

v3 alpha development

### CODE REVIEW

File: v3/generator/client/client.tmpl [hunk 1]
```diff
@@ -19,10 +19,14 @@ type Client struct {
...
-	middlewares     []RequestMiddlewareFn
+
+	// A list of callbacks for modifying requests which are generated before sending over
+	// the network.
+	requestEditors []RequestEditorFn
 }
 
-type RequestMiddlewareFn func(ctx context.Context, req *http.Request) error
+// RequestEditorFn  is the function signature for the RequestEditor callback function
+type RequestEditorFn func(ctx context.Context, req *http.Request) error
```
Comment (aureliar8) [inline v3/generator/client/client.tmpl:29]: I'm not 100% convinced with the name, perhaps we can keep the previous RequestInterceptor name even though the real behavior changed a bit (impossible to act on the response) ?

File: v3/generator/operations/operations.go [hunk 2]
```diff
@@ -30,22 +30,36 @@ func Generate(doc libopenapi.Document, path, packageName string) error {
...
 	"github.com/go-playground/validator/v10"
+	"github.com/google/uuid"
 )
 `, packageName))
 
+	h, err := os.ReadFile("./operations/helpers.tmpl")
+	if err != nil {
+		return err
+	}
+	output.Write(h)
```
Comment (aureliar8) [inline v3/generator/operations/operations.go:59]: I find it slightly weird to have a `.tmpl` that is actually not templated but pure go code.

**Review by aureliar8 — APPROVED** [review]

File: v3/generator/operations/operations.go (hunk 2 above)
Comment (pierre-emmanuelJ) [reply v3/generator/operations/operations.go]: https://github.com/exoscale/egoscale/pull/611/commits/81a4a65fdc9965f271ee57d10db2276929a28c6e

File: v3/generator/client/client.tmpl (hunk 1 above)
Comment (pierre-emmanuelJ) [reply v3/generator/client/client.tmpl]: https://github.com/exoscale/egoscale/pull/611/commits/f4c120bec71db146ecb71061a30f35cfba191e03

**Review by sauterp — APPROVED** [review]
thanks!

**Pull request merged**